plugins/coverimage.koplugin/main.lua

```diff
@@ -158,16 +197,440 @@ function CoverImage:onReaderReady(doc_settings)
+    Can used for migrating the contents of the old path to the new one
+]]
+function CoverImage:choosePathFile(touchmenu_instance, setting, folder_only, new_file, migrate)
+    local old_path, old_name = util.splitFilePathName(self[setting]) -- luacheck: no unused
+    UIManager:show(PathChooser:new{
+        select_directory = true,
+        select_file = not folder_only,
+        height = Screen:getHeight(),
+        path = old_path,
+        onConfirm = function(dir_path)
+            local mode = lfs.attributes(dir_path, "mode")
+            if folder_only then -- just select a folder
+                if not dir_path:find("/$") then
+                    dir_path = dir_path .. "/"
+                end
+                if migrate then
+                    migrate(self, self[setting], dir_path)
+                end
+                self[setting] = dir_path
+                G_reader_settings:saveSetting(setting, dir_path)
+                if touchmenu_instance then
+                    touchmenu_instance:updateItems()
+                end
+            elseif new_file and mode == "directory" then -- new filename should be entered or a file could be selected
+                local file_input
+                file_input = InputDialog:new{
+                    title =  _("Append filename"),
+                    input = dir_path .. "/",
+                    buttons = {{
+                        {
+                            text = _("Cancel"),
+                        },
+                        {
+                            text = _("Save"),
+                            callback = function()
+                                local file = file_input:getInputText()
+                                if migrate and self[setting] and self[setting] ~= "" then
+                                    migrate(self, self[setting], file)
+                                end
+                                self[setting] = file
+                                G_reader_settings:saveSetting(setting, file)
+                                if touchmenu_instance then
+                                    touchmenu_instance:updateItems()
+                                end
+                                UIManager:close(file_input)
+                            end,
+                        },
+                    }},
+                }
+                UIManager:show(file_input)
+                file_input:onShowKeyboard()
+            elseif mode == "file" then   -- just select an existing file
+                if migrate then
+                    migrate(self, self[setting], dir_path)
+                end
+                self[setting] = dir_path
+                G_reader_settings:saveSetting(setting, dir_path)
+                if touchmenu_instance then
+                    touchmenu_instance:updateItems()
+                end
+            end
+        end,
+    })
+end
+
+--[[--
+changes a G_reader_setting with a size spinner
```

Comment:
nit: `key` ;).

Reply:
Actually, better reword it, because that would be confusing.

e.g.,

Update a specific G_reader_setting's value via a Spinner